File: caoguodong/Remove Element.java
public class Solution {
    public int removeElement(int[] nums, int val) {
        if(nums == null || nums.length == 0){
            return 0;
        }
        int length = -1;
        for(int i=0;i<nums.length;i++){
            if(nums[i] == val && length < 0){
                length = i;
            }
            
            if(nums[i] != val && length >=0){
                int par = nums[i];
                nums[i] = nums[length];
                nums[length] = par;
                length++;
            }
        }
        if(length < 0){
            return nums.length;
        }
        return length;
    }
}

File: mayu/RemoveElement.swift
import Foundation

/*
 Given an array and a value, remove all instances of that value in place and return the new length.
 
 Do not allocate extra space for another array, you must do this in place with constant memory.
 
 The order of elements can be changed. It doesn't matter what you leave beyond the new length.
 
 Example:
 Given input array nums = [3,2,2,3], val = 3
 
 Your function should return length = 2, with the first two elements of nums being 2.
 */
class Solution {
    func removeElement(_ nums: inout [Int], _ val: Int) -> Int {
        if nums.count == 0 {
            return 0
        }
        if nums.contains(val) {
            nums.remove(at: nums.index(of: val)!)
            return removeElement(&nums, val)
        } else {
            return nums.count
        }
    }
}
